src/plugins/chart_expressions/expression_partition_vis/common/expression_functions/waffle_vis_function.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { LegendDisplay, PartitionVisParams } from '../types/expression_renderers';
+import { prepareLogTable } from '../../../../visualizations/common/prepare_log_table';
+import { ChartTypes, WaffleVisExpressionFunctionDefinition } from '../types';
+import {
+  PARTITION_LABELS_FUNCTION,
+  PARTITION_LABELS_VALUE,
+  PARTITION_VIS_RENDERER_NAME,
+  WAFFLE_VIS_EXPRESSION_NAME,
+} from '../constants';
+import { strings } from './i18n';
+
+export const waffleVisFunction = (): WaffleVisExpressionFunctionDefinition => ({
+  name: WAFFLE_VIS_EXPRESSION_NAME,
+  type: 'render',
+  inputTypes: ['datatable'],
+  help: strings.getPieVisFunctionName(),
+  args: {
+    metric: {
+      types: ['vis_dimension'],
+      help: strings.getMetricArgHelp(),
+      required: true,
+    },
+    bucket: {
+      types: ['vis_dimension'],
+      help: strings.getBucketArgHelp(),
+    },
+    splitColumn: {
+      types: ['vis_dimension'],
+      help: strings.getSplitColumnArgHelp(),
+      multi: true,
+    },
+    splitRow: {
+      types: ['vis_dimension'],
+      help: strings.getSplitRowArgHelp(),
+      multi: true,
+    },
+    addTooltip: {
+      types: ['boolean'],
+      help: strings.getAddTooltipArgHelp(),
+      default: true,
+    },
+    legendDisplay: {
+      types: ['string'],
+      help: strings.getLegendDisplayArgHelp(),
+      options: [LegendDisplay.SHOW, LegendDisplay.HIDE, LegendDisplay.DEFAULT],
+      default: LegendDisplay.HIDE,
+    },
+    legendPosition: {
+      types: ['string'],
+      help: strings.getLegendPositionArgHelp(),
+    },
+    nestedLegend: {
```

Comment:
We recently merged a PR that we remove the nestedLegend option from waffles (it doesnt make any sense) https://github.com/elastic/kibana/pull/123208